lib/fs.js

```diff
@@ -465,11 +465,16 @@ fs.readFileSync = function(path, options) {
 
 // Used by binding.open and friends
 function stringToFlags(flag) {
-  // Only mess with strings
-  if (typeof flag !== 'string') {
+  // Integers are already good to go
+  if (Number.isInteger(flag)) {
```

Comment:
This check doesn't work as you think it does: the C++ layer requires flags to fit within 32 bits, this check here only requires it to be integral. Try with `0xfffffffff`, and you will see.

That's why in my PR I pass anything with typeof number to the C++ layer, it seems easier to let it do this check, because it does it correctly. Though you can put more code in here, checking the bounds futher.